packages/phone-number-privacy/signer/src/domain/quota/domainQuota.service.ts

```diff
@@ -0,0 +1,52 @@
+import { isSequentialDelayDomain, KnownDomain } from '@celo/identity/lib/odis/domains'
+import { checkSequentialDelay } from '@celo/phone-number-privacy-common/lib/domains/sequential-delay'
+import Logger from 'bunyan'
+import { Transaction } from 'knex'
+import { toSequentialDelayDomainState } from '../../common/domain/domainState.mapper'
+import { DomainState } from '../../database/models/domainState'
+import { updateDomainState } from '../../database/wrappers/domainState'
+import { UnsupportedDomainError } from '../unsupportedDomain.error'
+import { IDomainQuotaService } from './domainQuota.interface'
+
+export class DomainQuotaService implements IDomainQuotaService {
+  constructor(private logger: Logger) {}
+
+  public async doesHaveRemainingQuota(
+    domain: KnownDomain,
+    domainState: DomainState
+  ): Promise<boolean> {
+    if (isSequentialDelayDomain(domain)) {
+      return checkSequentialDelay(domain, Date.now(), toSequentialDelayDomainState(domainState))
+        .accepted
+    } else {
+      throw new UnsupportedDomainError()
+    }
+  }
+
+  public async increaseQuotaCount(
+    domain: KnownDomain,
+    domainState: DomainState,
+    trx: Transaction<DomainState>
+  ): Promise<boolean> {
+    if (isSequentialDelayDomain(domain)) {
+      const result = checkSequentialDelay(
+        domain,
+        Date.now(),
+        toSequentialDelayDomainState(domainState)
+      )
+      if (!result.state) {
+        return false
+      }
+
+      return updateDomainState(
+        domain,
+        trx,
+        result.state.counter!,
+        result.state.timer!,
+        this.logger
+      ).then(() => true)
+    } else {
+      throw new UnsupportedDomainError()
+    }
```

Comment:
Similar to what I did in my PR, exhaustive case handling can be checked at compile-time using a canary function. Adding the canary check will allow this be treated as an implementation error. With this, we also would not need to add `UnsupportedDomainError`.
```suggestion
    }
    // canary provides a compile-time check that all subtypes of KnownDomain have branches. If a case
    // was missed, then an error will report that domain cannot be assigned to type `never`.
    const canary = (x: never) => x
    canary(domain)
    throw new Error('Implementation error. Input of type KnownDomain was not recognized')
```
https://github.com/celo-org/celo-monorepo/blob/f4dd249e9cef6a541b909c3dc7ae01888b088de2/packages/sdk/identity/src/odis/domains.ts#L136-L140